service/controller/controllercontext/spec.go

```diff
@@ -45,6 +48,7 @@ type ContextSpecTenantClusterTCCPAvailabilityZoneRouteTablePublic struct {
 
 type ContextSpecTenantClusterTCNP struct {
 	AvailabilityZones []ContextSpecTenantClusterTCNPAvailabilityZone
+	SecurityGroups    []*ec2.SecurityGroup
```

Comment:
should spec use ec2.SecurityGroup for eaiser handling or use just simple struct with `string ID` ? unsure

Reply:
A string slice is fine as long as this is maybe explained in a comment and we can comprehend what happens in the resource implementation filling the data in. Your call.